S3M Doc updates (#1970)
SumoLogic/sumologic-documentation: docs/send-data/opentelemetry-collector/troubleshooting-faq.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> This document contains common troubleshooting scenarios and frequently asked questions about Sumo Logic OpenTelemetry Collector.

Reviewer: ```suggestion
This document contains common troubleshooting scenarios and frequently asked questions about the Sumo Logic OpenTelemetry Collector.
```

The collector caches data when it’s unable to send it to the Sumo backend at the rate the data it is being produced. This can be due to various reasons, such as network issues or temporary unavailability of Sumo, and others.

#### Does the Sumo Logic backend know when a OpenTelemetry Collector is down? How does that work?

The collector sends a heartbeat message every 15 seconds (by default).

#### Does the OpenTelemetry Collector get throttled? Under what circumstances does this happen?

The behavior is the same as for any other collection method.

> * [Logs Throttling](https://help.sumologic.com/docs/manage/ingestion-volume/log-ingestion/#log-throttling)
> * [Metrics Throttling](https://help.sumologic.com/docs/metrics/manage-metric-volume/metric-throttling/)

Reviewer: ```suggestion
* [Logs Throttling](/docs/manage/ingestion-volume/log-ingestion/#log-throttling)
* [Metrics Throttling](/docs/metrics/manage-metric-volume/metric-throttling/)
```

## Collector connection failure

#### How do I provide a proxy setting for the Sumo Logic OpenTelemetry Collector to connect to the Sumo Logic backend?

If your Collector fails to connect to Sumo Logic, you may need to configure a proxy setting for the Sumo Logic OpenTelemetry Collector so that it can connect to the Sumo Logic backend. The information how to install the proxy has been provided on the Install Collector page for the specific platform.

> * [Linux Proxy](https://help.sumologic.com/docs/send-data/opentelemetry-collector/install-collector-linux/#using-proxy)
> * [Windows Proxy](https://help.sumologic.com/docs/send-data/opentelemetry-collector/install-collector-windows/#using-proxy)
> * [MacOS Proxy](https://help.sumologic.com/docs/send-data/opentelemetry-collector/install-collector-macos/#using-proxy))

Reviewer: ```suggestion
* [Linux Proxy](/docs/send-data/opentelemetry-collector/install-collector-linux/#using-proxy)
* [Windows Proxy](/docs/send-data/opentelemetry-collector/install-collector-windows/#using-proxy)
* [macOS Proxy](/docs/send-data/opentelemetry-collector/install-collector-macos/#using-proxy)
```

However, note that some functionalities, such as certain host metrics, may not be available when running the Collector as a non-root user.

> #### On Linux or MacOS, how can I view/edit configuration files within the `conf.d` directory?

Reviewer: ```suggestion
#### On Linux or macOS, how can I view/edit configuration files within the `conf.d` directory?
```

#### How do I uninstall the Sumo OpenTelemetry Collector?

> Refer to the Uninstall section in the below mentioned docs

Reviewer 2: ```suggestion
Refer to the Uninstall section in the following docs:
```

* [Linux](/docs/send-data/opentelemetry-collector/install-collector-linux/#uninstall)

> * [Windows]/docs/send-data/opentelemetry-collector/install-collector-windows/#uninstall)

Reviewer: ```suggestion
* [Windows](/docs/send-data/opentelemetry-collector/install-collector-windows/#uninstall)
```

> * [Linux](https://help.sumologic.com/docs/send-data/opentelemetry-collector/install-collector-linux/#uninstall)
> * [Windows](https://help.sumologic.com/docs/send-data/opentelemetry-collector/install-collector-windows/#uninstall)
> * [MacOS](https://help.sumologic.com/docs/send-data/opentelemetry-collector/install-collector-macos/#uninstall)

Reviewer: ```suggestion
* [Linux](/docs/send-data/opentelemetry-collector/install-collector-linux/#uninstall)
* [Windows]/docs/send-data/opentelemetry-collector/install-collector-windows/#uninstall)
* [macOS](/docs/send-data/opentelemetry-collector/install-collector-macos/#uninstall)
```

## Data ingestion and forwarding

#### How do I ingest historical data​?

> To ingest old or historical data use a [Filelog Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver) for log collection. By default, the Filelog Receiver starts reading logs from the end of the file. However, this can be changed by adjusting the configuration file to start reading from the beginning of the file by setting `start_at: beginning` in the receiver section. Below is an example

Reviewer: ```suggestion
To ingest old or historical data use a [Filelog Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver) for log collection. By default, the Filelog Receiver starts reading logs from the end of the file. However, this can be changed by adjusting the configuration file to start reading from the beginning of the file by setting `start_at: beginning` in the receiver section. Here is an example:
```

<img src={useBaseUrl('img/send-data/error-faq.png')} alt="error-faq.png" width="950" />

In this case, this means that the collector couldn’t find the file you wanted to collect logs from. There can be lot of reasons why this error happens, including not having appropriate permissions. 

> _Note: that collector will still collect & send data from other sources as long as there is no issues with those sources._

Reviewer: ```suggestion
:::note
The collector will still collect and send data from other sources as long as there is no issues with those sources.
:::
```

#### How can I filter out metrics from being collected?

> Both Linux and Windows users can utilize the filter processor provided by the OpenTelemetry Collector Contributor repository to filter out metrics from being collected. The filter processor can be used to selectively drop or pass through data based on certain conditions, but the specifics of the filtering would depend on the use case and requirements. For more information, refer to the [Filter Processor documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md.).

Reviewer 2: ```suggestion
Both Linux and Windows users can utilize the filter processor provided by the OpenTelemetry Collector Contributor repository to filter out metrics from being collected. The filter processor can be used to selectively drop or pass through data based on certain conditions, but the specifics of the filtering would depend on the use case and requirements. For more information, see [Filter Processor documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md.).
```

#### How can I filter out metrics from being collected?

> Both Linux and Windows users can utilize the filter processor provided by the OpenTelemetry Collector Contributor repository to filter out metrics from being collected. The filter processor can be used to selectively drop or pass through data based on certain conditions, but the specifics of the filtering would depend on the use case and requirements. For more information, see [Filter Processor documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md.).

Reviewer: ```suggestion
Both Linux and Windows users can utilize the filter processor provided by the OpenTelemetry Collector Contributor repository to filter out metrics from being collected. The filter processor can be used to selectively drop or pass through data based on certain conditions, but the specifics of the filtering would depend on the use case and requirements. For more information, see [Filter Processor documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md).
```
